navigation/perfect_memory.h

```diff
@@ -158,52 +160,50 @@ class PerfectMemoryRotater : public PerfectMemory<Store>
     template<class... Ts>
     auto getHeading(Ts &&... args) const
     {
-        calcImageDifferences(std::forward<Ts>(args)...);
+        auto rotater = Rotater::create(this->getUnwrapResolution(), this->getMaskImage(), std::forward<Ts>(args)...);
+        calcImageDifferences(rotater);
         const size_t numSnapshots = this->getNumSnapshots();
 
         // Now get the minimum for each snapshot and the column this corresponds to
-        std::vector<int> bestColumns(numSnapshots);
-        std::vector<float> minDifferences(numSnapshots);
+        std::vector<size_t> bestColumns;
```

Comment:
Maybe these could probably be made (mutable?) members to save  reallocations?

Reply:
Good idea!